pkg/policy/prefilter.go

```diff
@@ -0,0 +1,303 @@
+	fmt.Fprintf(fw, "#define CIDR4_LMAP_NAME %s\n", path.Base(p.Maps[prefixesV4Dyn].String()))
+	fmt.Fprintf(fw, "#define CIDR6_HMAP_NAME %s\n", path.Base(p.Maps[prefixesV6Fix].String()))
+	fmt.Fprintf(fw, "#define CIDR6_LMAP_NAME %s\n", path.Base(p.Maps[prefixesV6Dyn].String()))
+
+	if p.Config.fix4Enabled {
+		fmt.Fprintf(fw, "#define CIDR4_FILTER\n")
+		if p.Config.dyn4Enabled {
+			fmt.Fprintf(fw, "#define CIDR4_LPM_PREFILTER\n")
+		}
+	}
+	if p.Config.fix6Enabled {
+		fmt.Fprintf(fw, "#define CIDR6_FILTER\n")
+		if p.Config.dyn6Enabled {
+			fmt.Fprintf(fw, "#define CIDR6_LPM_PREFILTER\n")
+		}
+	}
+}
+
+func (p *PreFilter) dumpOneMap(which preFilterMapType, to []string) []string {
+	if p.Maps[which] == nil {
+		return to
+	}
+	return p.Maps[which].CIDRDump(to)
+}
+
+// Dump dumps revision and CIDRs as string slice of all participating maps
+func (p *PreFilter) Dump(to []string) ([]string, int64) {
+	p.Mutex.Lock()
+	defer p.Mutex.Unlock()
+	for i := prefixesV4Dyn; i < mapCount; i++ {
+		to = p.dumpOneMap(i, to)
+	}
+	return to, p.Revision
+}
+
+func (p *PreFilter) selectMap(ones, bits int) preFilterMapType {
+	if bits == 32 {
+		if ones == bits {
+			return prefixesV4Fix
+		}
+		return prefixesV4Dyn
+	} else if bits == 128 {
+		if ones == bits {
+			return prefixesV6Fix
+		}
+		return prefixesV6Dyn
+	} else {
+		return mapCount
+	}
+}
+
+// Insert inserts slice of CIDRs (doh!) for the latest revision
+func (p *PreFilter) Insert(revision int64, cidrs []net.IPNet) error {
+	var undoQueue []net.IPNet
+	var ret error
+
+	p.Mutex.Lock()
+	defer p.Mutex.Unlock()
+	if p.Revision != revision {
+		return fmt.Errorf("Latest revision is %d not %d", p.Revision, revision)
+	}
+	for _, cidr := range cidrs {
+		ones, bits := cidr.Mask.Size()
+		which := p.selectMap(ones, bits)
+		if which == mapCount || p.Maps[which] == nil {
+			ret = fmt.Errorf("No map enabled for CIDR string %s", cidr.String())
+			break
+		}
+		err := p.Maps[which].InsertCIDR(cidr)
+		if err != nil {
+			ret = fmt.Errorf("Error inserting CIDR string %s: %s", cidr.String(), err)
+			break
+		} else {
+			undoQueue = append(undoQueue, cidr)
+		}
+	}
+	if ret == nil {
+		p.Revision++
+		return ret
+	}
+	for _, cidr := range undoQueue {
+		ones, bits := cidr.Mask.Size()
+		which := p.selectMap(ones, bits)
+		p.Maps[which].DeleteCIDR(cidr)
+	}
+	return ret
+}
+
+// Delete deletes slice of CIDRs (doh!) for the latest revision
+func (p *PreFilter) Delete(revision int64, cidrs []net.IPNet) error {
+	var undoQueue []net.IPNet
+	var ret error
+
+	p.Mutex.Lock()
+	defer p.Mutex.Unlock()
+	if p.Revision != revision {
+		return fmt.Errorf("Latest revision is %d not %d", p.Revision, revision)
+	}
+	for _, cidr := range cidrs {
+		ones, bits := cidr.Mask.Size()
+		which := p.selectMap(ones, bits)
+		if which == mapCount || p.Maps[which] == nil {
+			ret = fmt.Errorf("No map enabled for CIDR string %s", cidr.String())
```

Comment:
You can immediately return here, no need for the break

Reply:
Yeah, since we don't modify anything yet, fixed.